triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceConfig.scala

```diff
@@ -64,7 +112,19 @@ object ServiceConfig {
         c.copy(commandTtl = Duration.ofSeconds(t))
       }
       .text("TTL in seconds used for commands emitted by the trigger. Defaults to 30s.")
+
+    opt[Map[String, String]]("jdbc")
+      .action((x, c) =>
+        c.copy(jdbcConfig = Some(JdbcConfig.create(x).fold(e => sys.error(e), identity))))
+      .optional()
+      .valueName(JdbcConfig.usage)
+      .text("JDBC configuration parameters. If omitted the service runs without a database.")
+
+    cmd("init-db")
```

Comment:
This is a bit confusing. You still require all the parameters that are required if you do not use the command due to the way `scopt` works so you have to pass in the ledger-host even though it is not used.

Reply:
Agreed. The problem I ran into is that the `TriggerDao` object requires an implicit execution context (for connecting to the database), and that is obtained from the actor system which is initialized with those config parameters.. Perhaps I could instead use the global execution context in the `init-db` case?

Reply:
Or a custom one. I never really understood which execution context you are supposed to use tbh :slightly_smiling_face:. Happy to keep it for now.